components/Dialogs/tests/snapshot/MDCAlertControllerDynamicTypeTests.m

```diff
@@ -0,0 +1,283 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MaterialSnapshot.h"
+
+#import <CoreGraphics/CoreGraphics.h>
+#import <UIKit/UIKit.h>
+
+#import "MaterialDialogs.h"
+#import "MaterialTypography.h"
+
+/** A @c MDCAlertController test fake to override the @c traitCollection to test for dynamic type.
+ */
+@interface AlertControllerDynamicTypeSnapshotTestFake : MDCAlertController
+@property(nonatomic, strong) UITraitCollection *traitCollectionOverride;
+@end
+
+@implementation AlertControllerDynamicTypeSnapshotTestFake
+
+- (UITraitCollection *)traitCollection {
+  return self.traitCollectionOverride ?: [super traitCollection];
+}
+
+@end
+
+@interface MDCAlertControllerDynamicTypeTests : MDCSnapshotTestCase
+@property(nonatomic, strong, nullable) AlertControllerDynamicTypeSnapshotTestFake *alertController;
+@end
+
+@implementation MDCAlertControllerDynamicTypeTests
+
+- (void)setUp {
+  [super setUp];
+
+  // Uncomment below to recreate all the goldens (or add the following line to the specific
+  // test you wish to recreate the golden for).
+  //  self.recordMode = YES;
+
+  self.alertController = [[AlertControllerDynamicTypeSnapshotTestFake alloc] init];
+  self.alertController.title = @"Material";
+  self.alertController.message =
+      @"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt "
+      @"ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation "
+      @"ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in "
+      @"reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur "
+      @"sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id "
+      @"est laborum.";
+  MDCAlertAction *fakeAction = [MDCAlertAction actionWithTitle:@"Foo"
+                                                       handler:^(MDCAlertAction *action){
+                                                       }];
+  [self.alertController addAction:fakeAction];
+  self.alertController.mdc_adjustsFontForContentSizeCategory = YES;
+  self.alertController.adjustsFontForContentSizeCategoryWhenScaledFontIsUnavailable = NO;
```

Comment:
Should we be setting `adjustsFontForContentSizeCategoryWhenScaledFontIsUnavailable` before `mdc_adjustsFontForContentSizeCategory` to reduce side effects?

Reply:
Updated throughout.